src/libpsl-native/src/getgrgid.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! @brief returns the groupname for a gid
+
+#include "getgrgid.h"
+
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <grp.h>
+#include <string.h>
+#include <unistd.h>
+
+//! @brief GetGrGid returns the groupname for a gid
+//!
+//! GetGrGid
+//!
+//! @param[in] gid
+//! @parblock
+//! The group identifier to lookup.
+//! @endparblock
+//!
+//! @retval groupname as UTF-8 string, or NULL if unsuccessful
+//!
+char* GetGrGid(gid_t gid)
+{
+    int32_t ret = 0;
+    struct group grp;
+    struct group* result = NULL;
+    char* buf;
+
+    int buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
+    if (buflen < 1)
+    {
+        buflen = 2048;
+    }
+
+allocate:
+    buf = (char*)calloc(buflen, sizeof(char));
+
+    errno = 0;
+    ret = getgrgid_r(gid, &grp, buf, buflen, &result);
+
+    if (ret != 0)
+    {
+        if (errno == ERANGE)
+        {
+            free(buf);
+            buflen *= 2;
+            goto allocate;
+        }
```

Comment:
what are you attempting to do with this check? `ERANGE` is only returned if the buffer is too small

Reply:
Upper boundary. Or we can support the buffer without upper limit?

Reply:
we should support whatever sysctl says it is. This can be set in the OS, while I don't believe there are any OSs which have larger than this, I don't _know_.